### PR TITLE
#175494041 Seting up mocha and chai for testing

### PR DESCRIPTION
**Pull request by tigthor**
### What does this PR do?
- Adding mocha,chai,sinon-chai and chai-http
 - writing simple initial test
 - running first test

### Description of PR to be completed
Rubbing test with mocha and chai

### How should this be manually tested
- clone this repo by running `git clone https://github.com/Soma-Technologies-Inc/visitAfrica-backend.git` in your terminal
- run `git checkout ch-settingmocha-175494041` to navigate to my branch
- then run `git pull` to make sure you have updated code from my branch
- then run `npm run test`
:open_mouth: you have run your first test

### Any background context you want to provide?
N/A
### What are the relevant pivotal tracker stories?
[#175494041](https://www.pivotaltracker.com/story/show/175494041)

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "presets": ["@babel/preset-env"]
+}
```
Comment (Niyonsengaeric) [inline .babelrc]: add a new line.

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,3 @@
 node_modules/
 package-lock.json
+.nyc_output/
```
Comment (Niyonsengaeric) [inline .gitignore]: add a new line.

**Review by Niyonsengaeric — CHANGES_REQUESTED** [review]

File: src/app.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+/* eslint-disable indent */
```
Comment (erickyvand) [inline src/app.js]: Instead of commenting the `eslint rule` try to correct that issue

File: package.json [hunk 4]
```diff
@@ -34,6 +34,16 @@
     "eslint-plugin-mocha": "^8.0.0",
     "eslint-plugin-node": "^11.1.0",
     "eslint-plugin-standard": "^4.1.0",
-    "nodemon": "^2.0.6"
+    "l": "^0.6.0",
+    "nodemon": "^2.0.6",
+    "nyc": "^15.1.0"
+  },
+  "dependencies": {
+    "body-parser": "^1.19.0",
```
Comment (erickyvand) [inline package.json]: remove this package because the recent version of `express` came with body parser

File: src/app.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+const port = process.env.PORT || 5000;
+
+app.get('/', (req, res) => res.status(200).send({
+        status: 200,
+        message: 'Hey there'
+    }));
+
+app.listen(port, () => {
+  console.log(`Server has started on port ${port}!!!`);
+});
```
Comment (erickyvand) [inline src/app.js]: @eustacheAki worked on this feature you can delete this file

File: src/test/test.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
+import chai from 'chai';
```
Comment (erickyvand) [inline src/test/test.js]: Folders should be plural, instead of `test` change it to `tests`

**Review by erickyvand — CHANGES_REQUESTED** [review]

File: .babelrc (hunk 1 above)
Comment (tigthor) [reply .babelrc]: I worked this feedback

File: .gitignore (hunk 2 above)
Comment (tigthor) [reply .gitignore]: Done

File: src/app.js (hunk 3 above)
Comment (tigthor) [reply src/app.js]: Done

File: package.json (hunk 4 above)
Comment (tigthor) [reply package.json]: Yes I worked on this

File: src/app.js (hunk 5 above)
Comment (tigthor) [reply src/app.js]: DOne

File: src/test/test.js (hunk 6 above)
Comment (tigthor) [reply src/test/test.js]: Done

**Pull request merged**